### PR TITLE
pipgrip 0.4.0 (new formula)

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/pipgrip.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  resource "pyparsing" do
+    url "https://files.pythonhosted.org/packages/c1/47/dfc9c342c9842bbe0036c7f763d2d6686bcf5eb1808ba3e170afdb282210/pyparsing-2.4.7.tar.gz"
+    sha256 "c203ec8783bf771a155b207279b9bccb8dea02d8f0c9e5f8ead507bc3246ecc1"
+  end
+
+  resource "setuptools" do
+    url "https://files.pythonhosted.org/packages/2f/8e/38259f4a44944a92068d5ff77230511a4c685604b47a81318f9e5cf2cc24/setuptools-49.2.0.zip"
+    sha256 "afe9e81fee0270d3f60d52608549cc8ec4c46dada8c95640c1a00160f577acf2"
+  end
```
Comment (SMillerDev) [inline Formula/pipgrip.rb:40]: i don't think this and wheel are needed, it was a desperate attempt to make the tests not fail on downloading resources.

Comment (dtrodrigues) [reply Formula/pipgrip.rb:40]: You're right - they're both installed by the `python@3.8.rb` formula.

**Comment by jonchang** [comment]
Yeah, I'm as baffled as you are.

So I got this to work on python2.7 but that's a nonstarter. Let me try with some other python versions. I'm almost certain that this is (ironically) some kind of version incompatibility brought on by some recent update to one of its dependencies.

**Comment by jonchang** [comment]
Looks like python3.7 + these resources works on a bare `pip install`:

pipgrip==0.4.0
anytree==2.8.0
six==1.15.0
click==7.1.2
packaging==20.4
pyparsing==2.4.7
pip==20.1.1
pkginfo==1.5.0.1
setuptools==49.2.0
wheel==0.34.2

**Comment by jonchang** [comment]
I wonder if this is a bug in pipgrip when it's being installed / run from a virtualenv. Will test in a docker image tomorrow.

**Comment by SMillerDev** [comment]
Checking with 3.7 now

**Comment by jonchang** [comment]
I can't reproduce this in Linux and for some reason it started working on my macOS install. We might have to install this the old way (`Language::Python.setup_install_args` and friends)

**Comment by jonchang** [comment]
Think I figured out the test.

**Comment by SMillerDev** [comment]
I think we should make an exception to the audit here on the grounds that we need the tool for an integration with brew.

**Comment by jonchang** [comment]
> I think we should make an exception to the audit here on the grounds that we need the tool for an integration with brew.

Yes, agreed. Either that or we ask all maintainers to star the repository 😈

**Review by jonchang — APPROVED** [review]

**Pull request closed without merging**